### DISCUSSION TITLE
Support for chapters

### DISCUSSION THREAD

**Discussion by lukassnake** [music-assistant/support#3061] [Feature requests and ideas]
Hi, I'm using Music Assistant (among other things) as an audiobook player to stream audiobooks. Many of my audiobooks are in a m4b file format, which contains chapters. In a normal audiobook player (or VLC for example), you can browse through these chapters. In MA, the book is displayed as a single long track. Is it possible to implement the functionality for chapters?

Thanks a lot for your excellent product!

**Comment by OzGav** [comment]
I don’t use audiobooks but as I understand it the chapters are shown in the now playing view and dots are shown on the timeline which coincides with the chapter start times. Is there anything else needed?

**Comment by lukassnake** [comment]
Yes, the audiobook (and chapters) support is now implemented ;-)

**Comment by rrooggiieerr** [comment]
Is there a special file structure needed for audio books?

**Reply by OzGav** [reply]
Refer to the docs for the provider you are using and then let us know if your question isn't answered https://www.music-assistant.io/music-providers/

**Reply by rrooggiieerr** [reply]
It's the file system provider, the documentation doesn't say anything about audio books. How does it know a file is (part of) an audio book vs a music album?

**Reply by OzGav** [reply]
I don't use this but they need to be in different shares and there is an option to specify what the share contains.

![image](https://github.com/user-attachments/assets/bf4d6e07-9dbf-4dcf-a211-46c57e6ad083)

**Reply by rrooggiieerr** [reply]
I think I figured it out, I have to add an additional file system provider for audio books
